### PR TITLE
Postgres adapter

### LINKED ISSUE #20: Use alternative NoSQL databases?
MongoDB is a great NoSQL database choice, but I had a few concerns:
1. I don't want to rely on MongoLab or another Database-as-a-service. I learn from experiences.
2. I don't want to install MongoDB on a compute instance and update versions, feed it RAM

I want to use either Google Datastore or Amazon DynamoDB. Both are good NoSQL database choices.
Google: https://cloud.google.com/datastore/
Amazon: https://aws.amazon.com/dynamodb/details/

I think it would be _nice_ to be able to interface with alternative NoSQL databases other than MongoDB.

### PR DESCRIPTION
**Pull request by drew-gross**
Ready to be merged after #2021 is merged

### CODE REVIEW

**Comment by facebook-github-bot-2** [comment]
@drew-gross updated the pull request.

**Comment by facebook-github-bot-1** [comment]
@drew-gross updated the pull request.

**Comment by facebook-github-bot-8** [comment]
@drew-gross updated the pull request.

**Comment by facebook-github-bot-2** [comment]
@drew-gross updated the pull request.

**Comment by facebook-github-bot-6** [comment]
@drew-gross updated the pull request.

**Comment by facebook-github-bot-9** [comment]
@drew-gross updated the pull request.

**Comment by facebook-github-bot-3** [comment]
@drew-gross updated the pull request.

**Comment by facebook-github-bot-6** [comment]
@drew-gross updated the pull request.

**Comment by facebook-github-bot-0** [comment]
@drew-gross updated the pull request.

**Comment by facebook-github-bot-3** [comment]
@drew-gross updated the pull request.

**Comment by facebook-github-bot-4** [comment]
@drew-gross updated the pull request.

**Comment by facebook-github-bot-9** [comment]
@drew-gross updated the pull request.

**Comment by facebook-github-bot-9** [comment]
@drew-gross updated the pull request.

**Comment by facebook-github-bot-9** [comment]
@drew-gross updated the pull request.

**Comment by drew-gross** [comment]
@flovilmart said it was OK to merge :D

File: src/Adapters/Storage/Postgres/PostgresStorageAdapter.js [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+  }
+
+  createClass(className, schema) {
+    return this._client.query('CREATE TABLE $<className:name> ()', { className })
+    .then(() => this._client.query('INSERT INTO "_SCHEMA" ("className", "schema", "isParseClass") VALUES ($<className>, $<schema>, true)', { className, schema }))
+  }
+
+  addFieldIfNotExists(className, fieldName, type) {
+    // TODO: Doing this in a transaction is probably a good idea.
+    return this._client.query('ALTER TABLE "GameScore" ADD COLUMN "score" double precision', { className, fieldName })
```
Comment (flovilmart) [inline src/Adapters/Storage/Postgres/PostgresStorageAdapter.js]: Uhm...

**Pull request merged**